hca-to-scea/helpers/check_experimental_design.py

```diff
@@ -84,9 +84,11 @@ def check_species_eligibility(xlsx_dict):
 
     species_list = []
     for biomaterial in biomaterial_tab:
-        species_key = "%s.genus_species.ontology_label" % (biomaterial)
-        species_list.extend(list(xlsx_dict[biomaterial][species_key].values))
+        if biomaterial in xlsx_dict.keys():
+            species_key = "%s.genus_species.ontology_label" % (biomaterial)
+            species_list.extend(list(xlsx_dict[biomaterial][species_key].values))
     species_list = list(set(species_list))
+    species_list = [x for x in species_list if str(x) != 'nan']
```

Comment:
Have you considered using [`dropna`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.dropna.html)?

I suggest to do a [pandas tutorial](https://pandas.pydata.org/docs/user_guide/10min.html) and search the [user guide](https://pandas.pydata.org/docs/user_guide/index.html) when you try to do something.
It is a very powerful library that helps you transform you date in many ways and by writing little code. I noticed that on occasion you are re-implementing stuff that pandas gives out of the box.

Reply:
Hi Amnon, in this specific case I did not try the pandas dropna function, but I'll have a go! I have tried using this function previously and found many times it didn't work on certain input types. But good point, I'll go through the pandas tutorial too

Reply:
Hi Amnon, in this specific case I did not try the pandas dropna function, but I'll have a go! I have tried using this function previously and found many times it didn't work on certain input types. But good point, I'll go through the pandas tutorial too

Reply:
Ok, I see what you mean, I have made the relevant changes with fewer lines of code.